src/main/java/org/semux/vm/client/SemuxPrecompiledContracts.java

```diff
@@ -64,6 +64,9 @@ public long getGasForData(byte[] data) {
                 AccountState as = semuxTrack.getAccountState();
                 DelegateState ds = semuxTrack.getDelegateState();
                 byte[] from = context.getCaller();
+                if (from == null) {
```

Comment:
I'd rather we provide an empty `from` address for local invocation, rather than allowing null pointer passed to the vm. Essentially, the local is similar to a transaction and the only differences are:
- No signing required;
- State changes will be discarded.

Reply:
Yea. I can make it include all the eth params.  I didn't know about "gas: QUANTITY - (optional) Integer of the gas provided for the transaction execution. eth_call consumes zero gas, but this parameter may be needed by some executions." And the like.

It does mean we'll need a separate fix for precompiled contracts calling local

I thought from was not assignable, only derived from sig.. but could be misremembering